### PR TITLE
New: import list source stashdb favorites

### LINKED ISSUE #134: Add Stashapp favorite performers as a list source
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

It is not.

### Describe the solution you'd like

I'd like to be able to favorite a performer and have them added as a list source in Whisparr. This would be similar to the way that Radarr/Sonarr integrate with Plex and the watchlists in Plex. If I add a show or movie to my watchlist in Plex, the *arr apps will add that to my download queue as it becomes available. This would work in a similar way.

### Describe alternatives you've considered

Open the performer's TPDB page to get the ID and manually create a list.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by enymawse**
#### Database Migration
No

#### Description
A user that has an account on stashdb.org with access to their API Key can add a new type of Import List. This new list titled "StashDB Favorites" can be configured with their API Key to list either their favorite performers, studios, or both. The users are provided the ability to not immediately download everything once the list has been added ([see discussion on issue #134](https://github.com/Whisparr/Whisparr/issues/134)). Additionally, the existing application limits the number of list items immediately available to download at 1000 ([see line 23 in HttpImportListBase.cs](https://github.com/Whisparr/Whisparr/blob/abe98a44b4582158b6a378971e70ab6cd240b4c5/src/NzbDrone.Core/ImportLists/HttpImportListBase.cs#L23)).

#### Screenshot (if UI related)
Import list looks like all the others, no frontend code was changed.

#### Todos
- [x] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [x] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Close #134

### CODE REVIEW

**Comment by sampulsar** [comment]
@enymawse 

If you are limited to the number of imported items you should sort the query by the created date descending, so new items are added.

**Comment by enymawse** [comment]
@sampulsar 

The query currently uses 'DATE' which appears to line up with Release date. I can see the case for having the query use both. What if I refactored it to give the user the option?

**Comment by enymawse** [comment]
@sampulsar 

Implemented a user selectable sort if you want to take a look at it. It's simple to walk it back to make the choice for the user. Let me know, happy to make the changes and squash commits.

**Comment by sampulsar** [comment]
@enymawse 

Great work squash the commits.

I can see another import that would be useful, if you could do one for a studio/performer, StahId, and set a release date so it will import scenes with a release dote after the entered date. Cam be done as a sperate PR after this one is promoted.

**Comment by enymawse** [comment]
> @enymawse
> 
> Great work squash the commits.
> 
> I can see another import that would be useful, if you could do one for a studio/performer, StahId, and set a release date so it will import scenes with a release dote after the entered date. Cam be done as a sperate PR after this one is promoted.

@sampulsar Thanks. Squashed. I can work on those. Will send up another PR when it's done.

**Comment by enymawse** [comment]
@sampulsar FYI

I looked into using release date and as it stands right now the Stash GraphQL endpoint doesn't support filtering by Date. I have a [PR #816](https://github.com/stashapp/stash-box/pull/816) open to resolve it. Until that gets merged and released I won't be able to create the additional import lists for Whisparr.

**Comment by enymawse** [comment]
@Tom-Estes @Qstick Any chance we can get approval to merge this into eros?

**Review by Tom-Estes — APPROVED** [review]

**Pull request merged**
